### PR TITLE
fix: Do not display Download actions on iOS Flagship app

### PR DESCRIPTION
**Pull request by Ldoppea**
Current implementation of iOS Flagship app doesn't handle files
download correctly

So we want to temporarily disable Download button on this platform

This commit is meant to be reverted when files download is fixed on iOS
Flagship app

---

Related PRs:

- https://github.com/cozy/cozy-ui/pull/2215

---

TODO:

- [ ] Upgrade cozy-ui after cozy/cozy-ui#2215 is merged (current upgrade commit doesn't have correct version and yarn.lock)

### CODE REVIEW

File: src/drive/web/modules/actions/useActions.jsx
Comment (trollepierre) [inline src/drive/web/modules/actions/useActions.jsx:28]: nit: it's not revert PR, it's revert commit https://github.com/cozy/cozy-drive/pull/2672/commits/1d6666ae0c14348e5630afe0f76373e28ac281e8
```suggestion
      // When fixed on this platform, revert PR #2672
```

File: package.json
Comment (trollepierre) [inline package.json:134]: it's not 71, it's 70

did you look at the breaking change of cozy-ui?
I think we should apply the codemods + remove @material-ui/core as dependency

File: src/drive/web/modules/actions/useActions.jsx
Comment (trollepierre) [inline src/drive/web/modules/actions/useActions.jsx:36]: i think this line is useless

**Comment by trollepierre** [comment]
I am quite afraid by the delta in the bundlemon report (2x + 13kb)

**Comment by Crash--** [comment]
> I am quite afraid by the delta in the bundlemon report (2x + 13kb)

Mainly because we add a lot of icons in UI. And Drive still uses the Sprite.

File: package.json
Comment (Ldoppea) [reply package.json:134]: Yep this is mentioned in the TODO list from the PR, i cannot put the final commit here yet as cozy-ui's PR is still not merged

The codemods have been applied already, not Text component is used in the project. Regarding material-ui, I don't get it, the BC is not that we MUST remove it, but it is about them not being mandatory anymore. If we must remove them, maybe we should rewrite the BC?

File: src/drive/web/modules/actions/useActions.jsx
Comment (Ldoppea) [reply src/drive/web/modules/actions/useActions.jsx:36]: Good catch, I put it before realizing that it was a push inside of a map. I'll remove it

File: src/drive/web/modules/actions/useActions.jsx
Comment (Ldoppea) [reply src/drive/web/modules/actions/useActions.jsx:28]: What about `// When fixed on this platform, revert this commit from PR #2672` ?

I cannot link to the commit because its SHA may change after the rebase/merge.

Comment (trollepierre) [reply src/drive/web/modules/actions/useActions.jsx:28]: Smart solution! Yes!

File: package.json
Comment (trollepierre) [reply package.json:134]: Yes it's not mandatory. => nit

But the maintainer of Drive will appreciate, not to have this useless dependency in the application.  😄

File: src/drive/web/modules/actions/index.jsx
Comment (Crash--) [inline src/drive/web/modules/actions/index.jsx:69]: You can do that differently and if you do that, you'll not have to adit the useActions hook. 

something like that should be enough: 
```
export const download = ({ client, vaultClient }) => {
  return isMobileApp()
    ? {
        icon: 'download',
        displayCondition: files => {
          if (isIOSApp()) return files.length === 1 && isFile(files[0])
          return files.reduce(
            (onlyFiles, file) => onlyFiles && isFile(file),
            true
          )
        },
        action: files => exportFilesNative(client, files, { vaultClient }),
        label: 'forwardTo',
        Component: function Download(props) {
          const { t } = useI18n()
          return (
            <ActionMenuItem
              onClick={() => {
                return exportFilesNative(client, props.files, { vaultClient })
              }}
              left={<Icon icon={isIOSApp() ? ShareIosIcon : ReplyIcon} />}
            >
              {t('SelectionBar.forwardTo')}
            </ActionMenuItem>
          )
        }
      }
    : {
        icon: 'download',
        displayCondition: files => {
        // ******* HERE is the change ****/ 
        if(isFlagshipApp() && isOS()) return false
          // We cannot generate archive for encrypted files, for now.
          // Then, we do not display the download button when the selection
          // includes an encrypted folder or several encrypted files
          return (
            !files.some(file => isEncryptedFolder(file)) &&
            !(files.length > 1 && files.some(file => isEncryptedFile(file)))
          )
        },
        action: files => downloadFiles(client, files, { vaultClient }),
        Component: function Download(props) {
          const { t } = useI18n()
          return (
            <ActionMenuItem
              onClick={() => {
                return downloadFiles(client, props.files, { vaultClient })
              }}
              left={<Icon icon={DownloadIcon} />}
            >
              {t('SelectionBar.download')}
            </ActionMenuItem>
          )
        }
      }
}
```

**Review by Crash-- — CHANGES_REQUESTED** [review]
see comment 🙏

File: package.json
Comment (trollepierre) [reply package.json:134]: As we decide changes on that import during last front meeting, I think it's better to let it go.

File: src/drive/web/modules/actions/index.jsx
Comment (Ldoppea) [reply src/drive/web/modules/actions/index.jsx]: How did I miss such an obvious method name? 😐 

Fixed in: https://github.com/cozy/cozy-drive/compare/cc0953ded5753e2e12e4172adbf11cfa67898622..5ae32366af08c739dcb09ecd56b524abd5b4b49b

**Review by Crash-- — APPROVED** [review]

**Pull request merged**
